nabd/sound_alsa.py

```diff
@@ -140,18 +145,7 @@ def _setup_device(self, device, channels, rate, width):
     device.setchannels(channels)
     device.setrate(rate)
 
-    # 8bit is unsigned in wav files
-    if width == 1:
-        device.setformat(alsaaudio.PCM_FORMAT_U8)
-    # Otherwise we assume signed data, little endian
-    elif width == 2:
-        device.setformat(alsaaudio.PCM_FORMAT_S16_LE)
-    elif width == 3:
-        device.setformat(alsaaudio.PCM_FORMAT_S24_3LE)
```

Comment:
Both exist, and the proper value is PCM_FORMAT_S24_3LE, not PCM_FORMAT_S24_LE.
Cf:
https://github.com/larsimmisch/pyalsaaudio/blob/master/alsaaudio.c#L2525
https://github.com/larsimmisch/pyalsaaudio/blob/master/playwav.py#L27
https://github.com/larsimmisch/pyalsaaudio/commit/40a1219dac5df9f314ac73f563cb1e77b9ea2782#diff-a9a41593ae6d535ab6b86fed0a75c022

Reply:
IDK, it seems like travis disagree:

```
E     AttributeError: module 'alsaaudio' has no attribute 'PCM_FORMAT_S24_3LE'
```

Reply:
Indeed, PCM_FORMAT_S24_3LE does not exist with 0.8.4. The constant was added since last release and requires pyalsaaudio@master.

PCM_FORMAT_S24_LE is not the proper constant for wav or mp3 files with width == 3 as explained in the pyalsaaudio commit referred to above.

Consequently, I suggest dropping case for width == 3.